### PR TITLE
small fixes

### PR DESCRIPTION
**Pull request by stokito**

### CODE REVIEW

**Comment by j256** [comment]
Good catch on the `==` @stokito but I'd rather fix it a different way by using the constant defined in `MagicEntryParser`.   I've pushed a fix for that.

**Comment by j256** [comment]
Thanks for the test changes.  Can you fix the indenting on a couple of them and I'll pull it.

**Comment by stokito** [comment]
But that's the same problem: you are comparing a String object as a reference. That was the main fix of the PR

**Comment by j256** [comment]
In this case I _want_ to compare the string as a reference.  If the entry has the unknown name as set by the parser then I want to detect it.

**Comment by stokito** [comment]
> In this case I want to compare the string as a reference.

the `equals()` method inside does the reference check so the performance will be almost the same. If your business logic is depends on the strict check by the reference then it's very confusing and fragile: it may be too easy to to break this contract during refactoring. 
So the explicit `equals()` call makes the code safer.

I fixed the ident  issue. Also I added few minor code style related changes.

**Comment by j256** [comment]
Your change can affects the logic.  I don't want the test to return true when the string happens to be the same characters as the `UNKNOWN_NAME` constant (in this case `"unknown"`).  I want it to happen _specifically_ when that particular string object is being used.  This is a well known pattern and is not "fragile" although I'll warrant that if you've not seen the pattern, you might be confused.  For example, if you have a queue and you want to put a `END_OF_DATA` type string in the queue, you use `==` instead of `.equals()` to ensure that your data doesn't contain your magic end string and cause your program to terminate early.

Another place that this is used is in Java interfaces when you want to define and test for a default return value.  `null` is not value for some language specific reason so you define a default string and use `==` when you want to say test for this specific object in which case I'll set it to be null.

File: src/main/java/com/j256/simplemagic/endian/EndianConverter.java [hunk 1]
```diff
@@ -12,17 +12,17 @@
 	 * 
 	 * @return The long or null if not enough bytes.
 	 */
-	public Long convertNumber(int offset, byte[] bytes, int size);
+	Long convertNumber(int offset, byte[] bytes, int size);
```
Comment (j256) [inline src/main/java/com/j256/simplemagic/endian/EndianConverter.java:15]: Please remove these.  My interfaces include the word public.  I know it's superfluous.

File: src/main/java/com/j256/simplemagic/entries/MagicEntry.java [hunk 2]
```diff
@@ -63,7 +63,7 @@
 	 */
 	ContentInfo matchBytes(byte[] bytes) {
 		ContentData data = matchBytes(bytes, 0, 0, null);
-		if (data == null || data.name == MagicEntryParser.UNKNOWN_NAME) {
+		if (data == null || data.name.equals(MagicEntryParser.UNKNOWN_NAME)) {
```
Comment (j256) [inline src/main/java/com/j256/simplemagic/entries/MagicEntry.java:66]: Please change these back to `==`.  This changes the logic.

File: src/main/java/com/j256/simplemagic/entries/MagicEntry.java [hunk 3]
```diff
@@ -197,7 +197,7 @@ private ContentData matchBytes(byte[] bytes, int prevOffset, int level, ContentD
 		 * NOTE: the children will have the first opportunity to set this which makes sense since they are the most
 		 * specific.
 		 */
-		if (name != MagicEntryParser.UNKNOWN_NAME && contentData.name == MagicEntryParser.UNKNOWN_NAME) {
+		if (!name.equals(MagicEntryParser.UNKNOWN_NAME) && contentData.name.equals(MagicEntryParser.UNKNOWN_NAME)) {
```
Comment (j256) [inline src/main/java/com/j256/simplemagic/entries/MagicEntry.java:200]: Please change these back to `==`.  This changes the logic.

File: src/main/java/com/j256/simplemagic/entries/MagicMatcher.java [hunk 4]
```diff
@@ -26,30 +26,30 @@
...
 	/**
 	 * Return the starting bytes of the pattern or null if none.
 	 */
-	public byte[] getStartingBytes(Object testValue);
+	byte[] getStartingBytes(Object testValue);
 
 	/**
 	 * Offset which we can update.
 	 */
-	public static class MutableOffset {
```
Comment (j256) [inline src/main/java/com/j256/simplemagic/entries/MagicMatcher.java:52]: Do you know what `static` on an inner class does?  Please look this up in your java manual and revert this change.

File: src/main/java/com/j256/simplemagic/entries/MagicType.java [hunk 5]
```diff
@@ -130,7 +130,7 @@
 		}
 	}
 
-	private MagicType(String name, MagicMatcher matcher) {
+	MagicType(String name, MagicMatcher matcher) {
```
Comment (j256) [inline src/main/java/com/j256/simplemagic/entries/MagicType.java:133]: Why remove private here?  Is it a bug?  Why make this change?

Comment (stokito) [reply src/main/java/com/j256/simplemagic/entries/MagicType.java:133]: as said in the commit message: enums constructors are already provate

**Comment by stokito** [comment]
ok so if this is not a bug and you are intentionally comparing strings by reference then basically that was the only one change that has some meaningful value.
It still looks not safe for me and brings some questions like "what about comparing two references from different classloaders"  and other fears because usually such comparings are really bugs.
Other changes was just intended to literally cleanup boilerplate from code.

Only e7125c3dd1b42f51f261e5808d086dfca8fb2eba has some additional value. Anyway I'll just close the PR as useless.
